### PR TITLE
[Checkbox] Design Guidelines

### PR DESCRIPTION
**Pull request by jesskuo4**
## Summary

Added `Checkbox` Design Guidelines

## List of notable changes:

- added new ComponentLayout to Checkbox docs
- added design guidelines

## What should reviewers focus on?

- 👀🙌  documentation: https://primer-d2984c5d07-26139705.drafts.github.io/components/Checkbox

## Supporting resources (related issues, external links, etc):

- Epic: https://github.com/github/primer/issues/2433
- Checkbox component: https://github.com/github/primer/issues/596
- [Brand] Checkbox: design guidelines: https://github.com/github/primer/issues/1996

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

File: apps/docs/content/components/Checkbox/index.mdx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+description: Use the checkbox component to select one or more options of a limited number of choices.
+---
+
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+
+## Anatomy
+
+![An image displaying a checkbox with a checkbox and label.](https://github.com/primer/brand/assets/912236/719022fa-4694-42c0-8554-b1923c65739f)
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:12]:
```suggestion
![An image displaying a checkbox with a checkbox and label.](https://github.com/primer/brand/assets/131988618/701f7c61-636e-4bca-bdee-c414043f525a)
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+Unlike the [Radio](/components/Radio) component or [Select](/components/Select) component, Checkbox does not restrict users to choosing just a single option. Instead, they necessitate a submission step, such as selecting an option or consent action. 
+
+Checkbox is commonly employed in forms to gather user input, granting them the ability to select zero or more options from a list.
+
+Keep the Checkbox component with direct labels, consistent styling (starting with capitalization and without ending in punctuation)
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/912236/eb8c5797-1704-4a92-8f9f-69c0b895f2b7" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:29]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/1687e5bd-8c81-4113-a2b5-60f2dcdcb4c0">
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        Ensure checkbox labels are direct. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/7890cbc2-e588-4399-b903-78f2e581d05b" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:35]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/92e2004a-d304-4988-aa4d-b8f8fead15fa">
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        Don't make labels confusing, avoid negative statements, and indirect. 
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/912236/eb8c5797-1704-4a92-8f9f-69c0b895f2b7" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:44]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/27a536e8-6bdb-4515-bb9c-029cd7deb482">
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        Use consistent styling, starting with capitalization, and not ending with punctuation.
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/7890cbc2-e588-4399-b903-78f2e581d05b" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:50]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/87460a05-c0e6-4d9b-89fe-56d1a889d0e8">
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        Don't have inconsistent styling, not starting with capitalization, or ending in punctuation.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/912236/eb8c5797-1704-4a92-8f9f-69c0b895f2b7" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:59]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/89fb2b6e-c178-4b6d-ba61-9205fc382770">
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        Use first person when asking users for acknowledgement or consent.
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/7890cbc2-e588-4399-b903-78f2e581d05b" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:65]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/e59627c9-d4f3-455d-8232-f08eef5a03b1">
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        Don't use third person when asking users for acknowledgement or consent.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/912236/eb8c5797-1704-4a92-8f9f-69c0b895f2b7" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:74]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/a788949c-0d08-4951-b9e2-6adf8d40c675">
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        When used in groups, order checkboxes in logical order, like smallest to largest or least restrictive to most restrictive. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/7890cbc2-e588-4399-b903-78f2e581d05b" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:80]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/0199ab3d-a199-468e-abba-8776b3a34c14">
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+description: Use the checkbox component to select one or more options of a limited number of choices.
+---
+
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+
+## Anatomy
+
+![An image displaying a checkbox with a checkbox and label.](https://github.com/primer/brand/assets/131988618/701f7c61-636e-4bca-bdee-c414043f525a)
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:12]:
```suggestion
![An image displaying a checkbox with a checkbox and label.](https://github.com/primer/brand/assets/131988618/61d1c706-2e3a-427c-bc37-0b77891b8007)
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+Unlike the [Radio](/components/Radio) component or [Select](/components/Select) component, Checkbox does not restrict users to choosing just a single option. Instead, they necessitate a submission step, such as selecting an option or consent action. 
+
+Checkbox is commonly employed in forms to gather user input, granting them the ability to select zero or more options from a list.
+
+Keep the Checkbox component with direct labels and consistent styling (starting with capitalization and without ending in punctuation)
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/1687e5bd-8c81-4113-a2b5-60f2dcdcb4c0" />
```
[the remark this answers is not in the document]
Comment (jesskuo4) [reply apps/docs/content/components/Checkbox/index.mdx:29]: Sounds great! Thank you so much for the feedback Josep! This is extremely appreciated :D

I'll have these images switched out, and I agree, for the Primer Brand components they don't have captions / descriptions 😄

File: apps/docs/content/components/Checkbox/index.mdx [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/89fb2b6e-c178-4b6d-ba61-9205fc382770" />
+    <Caption>
+        Use first-person when asking users for acknowledgement or consent.
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/e59627c9-d4f3-455d-8232-f08eef5a03b1" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:61]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/7225b935-b05d-46ef-9374-76a8404e8ebd" />
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        Don't use third-person when asking users for acknowledgement or consent.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/a788949c-0d08-4951-b9e2-6adf8d40c675" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:70]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/eecb8764-453d-474f-9e34-21d95c9aadef" />
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        When used in groups, order checkboxes in logical order, like smallest to largest or least restrictive to most restrictive. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/0199ab3d-a199-468e-abba-8776b3a34c14" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:76]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/9566ff9c-68e8-428e-967f-969edbd47139" />
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+    <Caption>
+        When possible, don't order checkboxes without logic.
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:78]:
```suggestion
        Don't use checkbox options that are exclusive to one another. Options should not be automatically deselected. 
```

Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:78]:
```suggestion
        Create checkbox options that are non-exclusive and flexible. 
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+description: Use the checkbox component to select one or more options of a limited number of choices.
+---
+
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+
+## Anatomy
+
+![An image displaying a checkbox with a checkbox and label.](https://github.com/primer/brand/assets/131988618/17a014ae-8d5f-4c52-890e-ecc49715da2b)
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:12]:
```suggestion
![An image displaying a checkbox with a checkbox and label.](https://github.com/primer/brand/assets/131988618/2152c720-5ba8-4832-b85b-0d54a8450207)

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+
+## Usage
+
+Use checkboxes to allow users to select multiple items from a list of individual items, or to mark one individual item as selected. Unlike the [Radio](/components/Radio) component or [Select](/components/Select) component, Checkbox does not restrict users to choosing just a single option. 
+
+Checkbox labels should be descriptive and concise. Write labels in sentence case.
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/3845d92c-0a76-4229-b961-a1442ec27646" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:25]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/45e4877b-345c-4a3d-904f-8478e5b8f9a4" />

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+    <Caption>
+        Checkbox labels should be descriptive and concise. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/afcba2b2-2754-40ec-9337-2734470faf08" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:31]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/b97d2f8b-4f60-44d4-b028-fd0981a7751a" />

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+    <Caption>
+        Don't make checkbox labels confusing, indirect, and avoid negative statements. 
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/a09bfb8f-7c7e-466a-8abc-e527cdce7d9b" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:40]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/8e29efd4-124f-4dc9-83c7-d80fd549c8da" />

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+    <Caption>
+        Use consistent styling (starting with capitalization and not ending with punctuation).
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/bf9feff5-bbdb-4414-8152-024652924a74" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:46]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/5d1c9068-cca5-484f-8573-047d4794f345" />

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+    <Caption>
+        Don't have inconsistent styling (not starting with capitalization or ending in punctuation).
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/43ae7520-8d25-43bc-b891-1e1abaebf44c" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:55]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/12c7391f-ea66-4aac-bd32-25cf49c75c9d" />

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+    <Caption>
+        Use first-person when asking users for acknowledgement or consent.
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/7225b935-b05d-46ef-9374-76a8404e8ebd" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:61]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/7e379318-3d12-4c99-b4ae-c33115791216" />
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+    <Caption>
+        Don't use third-person when asking users for acknowledgement or consent.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/eecb8764-453d-474f-9e34-21d95c9aadef" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:70]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/d68ff8bc-a8ba-4413-8d06-91a9102ee2d2" />

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+    <Caption>
+        Create checkbox options that are non-exclusive and allow for multiple selections. 
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/131988618/9566ff9c-68e8-428e-967f-969edbd47139" />
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:76]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/150d9724-425c-4ddc-bfc1-d368729abb95" />

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+  </Dont>
+</DoDontContainer>
+
+### States
+
+Checkboxes have static dimensions and four different states: rest (default), checked, indeterminate, and disabled. 
+
+An indeterminate checkbox state should be used if the input value is neither true nor false. This can be useful in situations where you are required to display an incomplete state, or one that is dependent on other input selections to determine a value.
+
+<img alt=""
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:89]:
```suggestion
<img alt="https://github.com/primer/brand/assets/131988618/89dbc496-2a83-44cf-81cd-9b6a82b4ba50" />

```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+/>
```
Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:90]:
```suggestion
```

File: apps/docs/content/components/Checkbox/index.mdx [hunk 27]
```diff
@@ -0,0 +1,103 @@
...
+    <img src="https://github.com/primer/brand/assets/131988618/7e379318-3d12-4c99-b4ae-c33115791216" />
+    <Caption>
+      Don't use third-person when asking users for acknowledgement or consent.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img src="https://github.com/primer/brand/assets/131988618/d68ff8bc-a8ba-4413-8d06-91a9102ee2d2" />
```
Comment (josepmartins) [inline apps/docs/content/components/Checkbox/index.mdx:70]: This image is empty 😱

Comment (josepmartins) [reply apps/docs/content/components/Checkbox/index.mdx:70]: let me know when it's fixed and I'll 👍 the PR so we can merge it 🥳

Comment (jesskuo4) [reply apps/docs/content/components/Checkbox/index.mdx:70]: Sounds good! thank you! 
Lemme do this :)

Comment (jesskuo4) [inline apps/docs/content/components/Checkbox/index.mdx:70]:
```suggestion
    <img src="https://github.com/primer/brand/assets/131988618/3a0a0eb1-372c-499b-9325-68337cef3428" />
```

Comment (josepmartins) [reply apps/docs/content/components/Checkbox/index.mdx]: I think something weird is happening (maybe its me?) but the updated image is empty too 😟

Comment (josepmartins) [reply apps/docs/content/components/Checkbox/index.mdx]: I fixed the image and updated the rest with 2x size for retina displays. This is ready to 🚢 !

**Review by josepmartins — APPROVED** [review]

**Pull request merged**